src/main/java/formflow/library/FileController.java

```diff
@@ -181,4 +213,117 @@ RedirectView delete(
+    byte[] fileData = cloudFile.getFileBytes();
+    long fileSize = cloudFile.getFileSize();
+
+    StreamingResponseBody responseBody = outputStream -> {
+      try {
+        outputStream.write(fileData);
+      } catch (IOException e) {
+        log.error("Error occurred while downloading file " + e.getMessage());
+      }
+    };
+
+    return ResponseEntity.ok()
+        .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=\"" + filename + "\"")
+        .contentType(MediaType.parseMediaType(contentType))
+        .contentLength(fileSize)
+        .body(responseBody);
+  }
+
+  /**
+   * @param httpSession  The current httpsession.
+   * @param submissionId The submissionId of the all the files that you would like to download.
+   * @return ON SUCCESS: A ResponseEntity with a status of ok, a header, and a responseBody that a zip folder with all the files
+   * assoicated with a submission.
+   * <p>ON FAILURE: A ResponseEntity with a status of forbidden-403 or not found- 404</p>
+   */
+  @GetMapping("/file-download/{submissionId}")
+  ResponseEntity<StreamingResponseBody> downloadAllFiles(
+      HttpSession httpSession,
+      @PathVariable String submissionId
+  ) {
+    Optional<Submission> maybeSubmission = submissionRepositoryService.findById(UUID.fromString(submissionId));
+    if (httpSession.getAttribute("id").toString().equals(submissionId) && maybeSubmission.isPresent()) {
+      if (maybeSubmission.isEmpty()) {
+        log.error(String.format("Session %d does not exist", submissionId));
+        return ResponseEntity.notFound().build();
+      }
+
+      Submission submission = maybeSubmission.get();
+      List<UserFile> userFiles = uploadedFileRepositoryService.findAllBySubmissionId(submission);
+      if (userFiles.isEmpty()) {
+        return ResponseEntity.notFound().build();
+      }
+
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      try (ZipOutputStream zos = new ZipOutputStream(baos)) {
+        for (UserFile userFile : userFiles) {
+          ZipEntry fileEntry = new ZipEntry(userFile.getOriginalName());
+          fileEntry.setSize(userFile.getFilesize().longValue());
+          zos.putNextEntry(fileEntry);
+
+          CloudFile cloudFile = cloudFileRepository.get(userFile.getRepositoryPath());
+          byte[] fileBytes = cloudFile.getFileBytes();
+          zos.write(fileBytes);
+          zos.closeEntry();
+        }
+      } catch (IOException e) {
+        log.error("Error occurred while downloading file " + e.getMessage());
+        return ResponseEntity.internalServerError().build();
+      }
+
+      StreamingResponseBody responseBody = outputStream -> {
+        baos.writeTo(outputStream);
+        baos.close();
+      };
+
+      return ResponseEntity.ok()
+          .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=\"" + "UserFiles-" + submission.getId() + ".zip" + "\"")
+          .body(responseBody);
+    } else {
+      log.error(
```

Comment:
Or the requested submission does not exist.... as per the maybeSubmission.isPresent() check being false. 

If the submission doesn't exist, it would be good to differentiate that in the error message we log.